### PR TITLE
update to ES 8 and add necessary adjustements to its config

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/issues/2332

### CODE REVIEW

File: php/containers.json [hunk 1]
```diff
@@ -499,10 +499,18 @@
...
+        "xpack.security.enabled=false",
+        "xpack.watcher.enabled=false",
+        "xpack.graph.enabled=false",
+        "xpack.ml.enabled=false",
+        "bootstrap.memory_lock=true",
+        "cluster.name=nextcloud-aio",
         "discovery.type=single-node",
-        "ES_JAVA_OPTS=-Xms1024M -Xmx1024M",
-        "POSTGRES_HOST=nextcloud-aio-database"
+        "thread_pool.write.queue_size=1000"
```
Comment (Zoey2936) [inline php/containers.json:513]: not sure if all of them are needed.. can you please check if xpack.security.enabled is enough?

File: php/containers.json [hunk 2]
```diff
@@ -499,10 +499,18 @@
       ],
       "internal_port": "9200",
       "environment": [
+        "POSTGRES_HOST=nextcloud-aio-database",
         "TZ=%TIMEZONE%",
+        "ES_JAVA_OPTS=-Xms1024M -Xmx1024M -Des.enforce.bootstrap.checks=true",
```
Comment (szaimen) [inline php/containers.json:504]: Check if we can change the loglevel to warn: https://www.elastic.co/guide/en/elasticsearch/reference/current/logging.html#configuring-logging-levels

File: Containers/fulltextsearch/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Probably from here https://github.com/elastic/elasticsearch/blob/main/distribution/docker/src/docker/Dockerfile
-FROM elasticsearch:7.17.10
+FROM elasticsearch:8.7.1
```
Comment (szaimen) [inline Containers/fulltextsearch/Dockerfile:2]:
```suggestion
FROM elasticsearch:8.8.0

```

**Comment by szaimen** [comment]
Apparently https://github.com/juliushaertl/nextcloud-docker-dev/blob/940602d00a5216b01b6fe35c0c817ef8f79a0de0/docker-compose.yml#L688-L702 might be something to take inspiration from

File: Containers/nextcloud/entrypoint.sh [hunk 4]
```diff
@@ -677,7 +677,7 @@ if [ "$FULLTEXTSEARCH_ENABLED" = 'yes' ]; then
         php /var/www/html/occ app:update files_fulltextsearch
     fi
     php /var/www/html/occ fulltextsearch:configure '{"search_platform":"OCA\\FullTextSearch_Elasticsearch\\Platform\\ElasticSearchPlatform"}'
-    php /var/www/html/occ fulltextsearch_elasticsearch:configure "{\"elastic_host\":\"http://$FULLTEXTSEARCH_HOST:9200\",\"elastic_index\":\"nextcloud-aio\"}"
+    php /var/www/html/occ fulltextsearch_elasticsearch:configure "{\"elastic_host\":\"http://elastic:$ELASTIC_PASSWORD@$FULLTEXTSEARCH_HOST:9200\",\"elastic_index\":\"nextcloud-aio\"}"
```
Comment (szaimen) [inline Containers/nextcloud/entrypoint.sh:680]: need to check if this works

Comment (Zoey2936) [reply Containers/nextcloud/entrypoint.sh]: does it not work without a password?

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: I suppose it does but with password is more secure...

**Comment by Zoey2936** [comment]
sorry, I can't review I'm on vaction until 28.07.

**Comment by Zoey2936** [comment]
> sorry, I can't review I'm on vaction until 28.07.

or better said not test things

File: Containers/nextcloud/entrypoint.sh [hunk 5]
```diff
@@ -677,7 +677,7 @@ if [ "$FULLTEXTSEARCH_ENABLED" = 'yes' ]; then
         php /var/www/html/occ app:update files_fulltextsearch
     fi
     php /var/www/html/occ fulltextsearch:configure '{"search_platform":"OCA\\FullTextSearch_Elasticsearch\\Platform\\ElasticSearchPlatform"}'
-    php /var/www/html/occ fulltextsearch_elasticsearch:configure "{\"elastic_host\":\"http://$FULLTEXTSEARCH_HOST:9200\",\"elastic_index\":\"nextcloud-aio\"}"
+    php /var/www/html/occ fulltextsearch_elasticsearch:configure "{\"elastic_host\":\"http://elastic:$FULLTEXTSEARCH_PASSWORD@$FULLTEXTSEARCH_HOST:9200\",\"elastic_index\":\"nextcloud-aio\"}"
```
Comment (szaimen) [inline Containers/nextcloud/entrypoint.sh:680]: Do this in a follow-up

File: php/containers.json [hunk 6]
```diff
@@ -499,10 +501,17 @@
       ],
       "internal_port": "9200",
       "environment": [
+        "POSTGRES_HOST=nextcloud-aio-database",
         "TZ=%TIMEZONE%",
+        "FULLTEXTSEARCH_PASSWORD=%FULLTEXTSEARCH_PASSWORD%",
```
Comment (szaimen) [inline php/containers.json:506]: Same

**Pull request merged**